encode.go

```diff
@@ -367,8 +367,14 @@ func ParseTimestamp(currentLocation *time.Location, str string) (time.Time, erro
 	timeSep := daySep + 3
 	day := p.mustAtoi(str, daySep+1, timeSep)
 
+	isBC := str[len(str)-3:] == " BC"
```

Comment:
This will panic if `len(str) < 3`. The previous code correctly checked for that.